rct_image_tools/src/rct_image_tools/image_observation_finder.cpp

```diff
@@ -321,15 +321,15 @@ static bool extractKeyPoints(const ObservationPoints& centers, ObservationPoints
 
 template <typename PARAMS, typename DETECTOR_PTR, typename DETECTOR>
 static bool extractModifiedCircleGrid(const cv::Mat& image, const rct_image_tools::ModifiedCircleGridTarget& target,
-                                      ObservationPoints& observation_points, const PARAMS* set_params = nullptr)
+                                      ObservationPoints& observation_points, const bool debug, const PARAMS* set_params = nullptr)
```

Comment:
I'm not a fan of adding a boolean flag to the object constructors to control their internal behavior. If feasible, I think it would be better to expose the debug image creation/display as a standalone function applications could call. That way you aren't exposing it to all users who will see the flag in the declaration and wonder what it does.

Reply:
I did a bit of an overhaul to the `CircleDetector` class. Let me know what you think